src/Views/AppInfoView.vala

```diff
@@ -929,6 +935,11 @@ namespace AppCenter.Views {
                 image.icon_name = "image-x-generic";
                 image.halign = Gtk.Align.CENTER;
                 image.gicon = pixbuf;
+                if (caption != null) {
+                    // AppStream spec says "ideally not more than 100 characters"
+                    int max_caption_len = 500;
+                    image.tooltip_text = ellipsize (caption, max_caption_len);
+                }
```

Comment:
Actually, does it make sense use GTK's [custom tooltip](https://valadoc.org/gtk+-3.0/Gtk.Tooltip.set_custom.html) functionality to add a label, and use GTK and Pango's [built-in ellipsization](https://valadoc.org/gtk+-3.0/Gtk.Label.ellipsize.html) here?

```suggestion

                if (caption != null) {
                    image.has_tooltip = true;
                    image.query_tooltip.connect ((x, y, keyboard_tooltip, tooltip) => {
                        tooltip.set_custom (new Gtk.Label (caption) {
                            ellipsize = Pango.EllipsizeMode.END,
                            // AppStream spec says "ideally not more than 100 characters"
                            max_width_chars = 200
                        });

                        return true;
                    });
                }
```

Reply:
I tried this out, but I think Label applies ellipsis to each line independently, i.e. it will cut off each line at 200 chars. On the one hand that is a bit long for a single line. On the other hand it doesn't prevent entire caption from being longer than 200 chars (if it has multiple lines). This is what it looks like to use the Label:

![label-ellipsize](https://user-images.githubusercontent.com/5374391/148281335-c23506d2-92a3-4194-89aa-984c0e3851ea.png)

I tried applies line wrapping on top of it, but it doesn't work in combination with ellipsizing. In the end I think manually cutting the caption at 200 chars total and using the default tooltip implementation (which limits the width by wrapping) gives the nicest looking result:

![tooltip-200](https://user-images.githubusercontent.com/5374391/148286447-1340c948-fb52-4a6c-b38b-1f411b3fec35.png)